code/game/machinery/fusion_engine.dm

```diff
@@ -352,6 +337,15 @@
 	return (fuel_amount == max_fuel_amount)
 
 /obj/item/fuelCell/proc/give(amount)
-	fuel_amount += amount
-	if(fuel_amount > max_fuel_amount)
-		fuel_amount = max_fuel_amount
+	fuel_amount = min(fuel_amount + amount, max_fuel_amount)
+	update_icon()
```

Comment:
if inside the generator you cant see it so it might be better to only update it when its ejected